flow/flow.go

```diff
@@ -1495,20 +1495,20 @@ func pFastGenerate(parameters interface{}, inIndex []int32, stopper [2]chan int,
 			err := low.AllocateMbufs(bufs, mempool, burstSize)
 			if err != nil {
 				low.ReportMempoolsState()
-				common.LogFatal(common.Debug, err)
-			}
-			if vector == false {
-				for i := range bufs {
-					// TODO Maybe we need to prefetch here?
-					tempPacket = packet.ExtractPacket(bufs[i])
-					generateFunction(tempPacket, context[0])
-					if reportMbits {
-						currentState.V.Bytes += uint64(tempPacket.GetPacketLen())
+			} else {
```

Comment:
You remove "currentState.V.Bytes += uint64(tempPacket.GetPacketLen())" line if there is no place in mempool. It has two problems:

1) inconsistent state for V.Bytes and V.Packets.
2) Scheduler will get wrong number of packets (as well as updatePortStats). It will not clone - that is true, but it behavior becomes less predictable.

Reply:
Yes, but currently megabits is already unimplemented for vector functions at all. I also think that it is ok for a corner case when generator is unable to allocate new packets. Either this situation quickly fixes itself, or the application is already deep in troubles and inconsistent behavior is the least of its problems.